### ISSUE TITLE
use as texture on a plane

### ISSUE THREAD

**Issue by neodev2** [gree/unity-webview#186]
Is it possible to use it as texture on some object like a plane in OS X?

**Comment by KojiNakamaru** [comment]
Yes, in OS X. For the sample app, if you place a "Sphere", modify WebViewObject.cs as below:
```diff
--- WebViewObject.cs~	2017-05-21 02:26:20.000000000 +0900
+++ WebViewObject.cs	2017-06-02 16:52:10.000000000 +0900
@@ -52,7 +52,7 @@
 #if UNITY_EDITOR_OSX || UNITY_STANDALONE_OSX
     IntPtr webView;
     Rect rect;
-    Texture2D texture;
+    public Texture2D texture;
     string inputString;
     bool hasFocus;
 #elif UNITY_IPHONE
```
and SampleWebView.cs as below:
```diff
diff --git a/sample/Assets/Scripts/SampleWebView.cs b/sample/Assets/Scripts/SampleWebView.cs
index 4311168..6b19418 100644
--- a/sample/Assets/Scripts/SampleWebView.cs
+++ b/sample/Assets/Scripts/SampleWebView.cs
@@ -104,6 +104,11 @@ public class SampleWebView : MonoBehaviour
             "   };" +
             "});");
 #endif
+        while (webViewObject.texture == null) {
+            yield return null;
+        }
+        var sphere = GameObject.Find("Sphere");
+        sphere.GetComponent<Renderer>().material.mainTexture = webViewObject.texture;
         yield break;
     }
``` 
You can get the following:

![image](https://cloud.githubusercontent.com/assets/2645978/26716636/371aeaa6-47b5-11e7-8287-167aea839761.png)

**Comment by neodev2** [comment]
I followed these steps, then I run play, but there is no sphere in my result like in your image... I have no console errors.. ??

**Comment by neodev2** [comment]
I am new to Unity, I followed your steps, now I have the Sphere following what happens in the WebViewObject, so if I scroll the page in WebViewObject, the same happens onto the Sphere. But if I try to interact directly with the sphere nothing happens.. I am not sure if this is because the sphere remains obstacled behind that WebViewObject.. I would like to have only the sphere and interact directly with it

**Comment by KojiNakamaru** [comment]
You need to implement any interaction with the sphere by yourself and send corresponding events to WebView. Please check `WebViewOjbect#OnGUI()`. It sends events to WebView so that it can update its internal state and the texture. Please also note `WebViewOjbect#OnGUI()` draws the texture onto the screen with `GUI.DrawTexture()`. If you commented out `GUI.DrawTexture()`, there will be only a sphere.

**Comment by topherfangio** [comment]
@KojiNakamaru While reading the code, I see the `WebViewObject#OnGUI()` is wrapped with `#if UNITY_EDITOR_OSX || UNITY_STANDALONE_OSX`.

Am I correct in understanding then, that the texture mapping is **not** available on iOS/Android but only the editor and OS X apps?

Is it possible to do it on iOS? That's what I really need. Thanks for all of the effort you put into this!

**Comment by KojiNakamaru** [comment]
Yes, your understanding is correct. For now this plugin doesn't support rendering webview into texture on mobile devices.
